### PR TITLE
Fixed ConnectionString issue

### PR DESCRIPTION
**Pull request by arvindgh**

### CODE REVIEW

File: Ginger/GingerCore/Environments/Database.cs [hunk 1]
```diff
@@ -255,61 +264,57 @@ public string GetConnectionString()
...
+            if (String.IsNullOrEmpty(ConnectionStringCalculated))
             {
-                connStr = ConnectionStringCalculated.Replace("{USER}", UserCalculated);
-     
-                String deCryptValue = EncryptionHandler.DecryptString(PassCalculated, ref res, false);
-                if (res == true)
-                    { connStr = connStr.Replace("{PASS}", deCryptValue); }
-                else
-                    { connStr = connStr.Replace("{PASS}", PassCalculated); }
+                connStr = CreateConnectionString();
```
Comment (ambarkhan) [inline Ginger/GingerCore/Environments/Database.cs]: Codacy Issue found: [Remove this useless assignment to local variable 'connStr'.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5217045)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5217045&bid=17105311)

File: Ginger/GingerCore/Environments/Database.cs [hunk 2]
```diff
@@ -255,61 +264,57 @@ public string GetConnectionString()
...
+            String strProvider;
+            ConnectionString = "Data Source=" + TNS + ";User Id={USER};Password={PASS};";
 
-                if (DBType == eDBTypes.MSAccess)
-                {
-                    if (strConnString.Contains(".accdb")) strProvider = "Provider=Microsoft.ACE.OLEDB.12.0;";
-                    else strProvider = "Provider=Microsoft.Jet.OLEDB.4.0;";
+            if (DBType == eDBTypes.MSAccess)
+            {
+                if (TNSCalculated.Contains(".accdb")) strProvider = "Provider=Microsoft.ACE.OLEDB.12.0;";
```
Comment (ambarkhan) [inline Ginger/GingerCore/Environments/Database.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5217045)

File: Ginger/GingerCore/Environments/Database.cs [hunk 3]
```diff
@@ -182,7 +182,10 @@ public string TNSCalculated
             get
             {
                 VE.Value = TNS;
-                return mVE.ValueCalculated;
+                mVE.DecryptFlag = true;
```
Comment (JinendraGhodke) [inline Ginger/GingerCore/Environments/Database.cs]: Add a common function which will take a string and return calculated value

File: Ginger/GingerCore/Environments/Database.cs [hunk 4]
```diff
@@ -255,61 +264,57 @@ public string GetConnectionString()
...
+                    ConnectionString = "Server=" + TNS + ";User Id={USER}; Password={PASS};Database=" + Name + ";";
+
                 }
-                
             }
-            ConnectionString = connStr;
-            return connStr;
+            else if (DBType == eDBTypes.MySQL)
+            {
+                ConnectionString = "Server=" + TNS + ";Database=" + Name + ";UID={USER};PWD={PASS}";
```
Comment (JinendraGhodke) [inline Ginger/GingerCore/Environments/Database.cs]: if password variable is used then TNS wont' be descrypted

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5217045&bid=17105311)

File: Ginger/GingerCore/Environments/Database.cs [hunk 5]
```diff
@@ -255,61 +255,62 @@ public string GetConnectionString()
...
+            if (String.IsNullOrEmpty(ConnectionStringCalculated))
             {
-                connStr = ConnectionStringCalculated.Replace("{USER}", UserCalculated);
-     
-                String deCryptValue = EncryptionHandler.DecryptString(PassCalculated, ref res, false);
-                if (res == true)
-                    { connStr = connStr.Replace("{PASS}", deCryptValue); }
-                else
-                    { connStr = connStr.Replace("{PASS}", PassCalculated); }
+                connStr = CreateConnectionString();
```
Comment (ambarkhan) [inline Ginger/GingerCore/Environments/Database.cs]: Codacy Issue found: [Remove this useless assignment to local variable 'connStr'.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5217045)

**Pull request merged**